__tests__/authButton.test.jsx

```diff
@@ -0,0 +1,30 @@
+import { screen, render } from "@testing-library/react";
+import AuthButton from "../components/authButton";
+import "@testing-library/jest-dom";
+import { useSession } from "next-auth/react";
+
+jest.mock('next-auth/react', () => {
```

Comment:
These codefactor warnings can be resolved by installing [eslint-plugin-jest](https://www.npmjs.com/package/eslint-plugin-jest)

Reply:
It's added now, they still seem to be there. Do we need some kind of config? Also is it possible to resolve the socket security warnings - or can only the jest library maintainers fix that?

Reply:
To get `eslint-plugin-jest` working you need to add to `.estlintrc.js`

```diff
  env: {
    node: true,
    browser: true,
    commonjs: true,
+   es6: true,
    'jest/globals': true,
  },
  extends: [
    'eslint:recommended',
    'next/core-web-vitals',
    'plugin:cypress/recommended',
+   'jest',
    'prettier'
  ],
```

should do the trick. Also, it would be better as dev dependency.

As for the socket warnings, they shouldn't be an issue since jest will never be run in production.

Reply:
Per the package's [documentation](https://www.npmjs.com/package/eslint-plugin-jest), I added 
`{
...
plugins:jest
...
}` 
into `eslintrc.js`. Then I extended the `eslint-plugin-jest` package using `extend:[plugin:jest/recommended]`. When I tried extending `jest`, I came across this error:

`Oops! Something went wrong! :(
ESLint: 8.34.0
ESLint couldn't find the config "jest" to extend from. Please check that the name of the config is correct.`

This lead me to go to the package's documentation and find that I needed to explicitly state the plugin and then extend the plugin.

The codefactor warnings are no longer there but I want to make sure that this is the intended setup since it is slightly different than what you mentioned.

Reply:
Sorry, @GuillermoFloresV, I'm not sure what I was thinking.  The simplest setting is just to extend `plugin:jest/recommended` like you have. There's no need to also declare `plugins: ['jest']` since that's included automatically when you extend `plugin:jest/recommended`.

One other thing I spotted: the `.jsx` files are ignored when you run `npm run lint:code`.  What I would recommend is to have `"lint:code": "eslint . --cache"` and add 

```
  overrides: [
    {
      files: ['*.js', '*.jsx'],
    }
  ],
```

to the root of your .eslintrc.js. That just tells `eslint` that you'd like to lint .jsx files as well as .js.

The reason I'd go this route is that (ironically) `--ext` overrides `overrides`. So it could be confusing if someone tried to use both in future. Also, since `--ext` is just more eslint configuration, it's simpler if it's all in one place, namely the .eslintrc.js file.